src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodec.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017 Rocana
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.rocana.lucene.codec.v1;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.PostingsFormat;
+import org.apache.lucene.codecs.lucene54.Lucene54Codec;
+import org.apache.lucene.codecs.perfield.PerFieldPostingsFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Custom codec that speeds up opening Searchers.
+ *
+ * We don't need Lucene's entire-file checksum since HDFS already
+ * uses checksums for HDFS blocks. Eliminating Lucene's checksum for
+ * the entire file improves Searcher open performance.
+ *
+ * See ROCANA-8229
+ */
+public class RocanaSearchCodec extends FilterCodec {
+
+  public static final String NAME = "RocanaSearchCodecV1";
```

Comment:
This line differs from the version in the main Rocana repo. We've moved the version number from the class to the package. So RocanaSearchCodecV1.java is now RocanaSearchCodec.java. But we still need unique 'short names', so rather than doing `getClass().getSimpleName()` we now hardcode the name and it differs from the class name.